### ISSUE TITLE
Pipeline Stopped Working

### ISSUE THREAD

**Issue by trevor-wu** [e-mission/e-mission-docs#289]
![screenshot_20190117-221735](https://user-images.githubusercontent.com/35517530/51368885-1c1bfb80-1aa6-11e9-8686-3f5d73a981f0.png)
The pipeline is not running on my version of the application. For example an entry from my .timeline dump is missing mode inferences.
My .timeline output for "data" looks like:
`"data": {
      "name": "USERCACHE",
      "ts": 1547712006.598315,
      "reading": 0.02276834473013878
    }`
while the .timeline output for someone else looks like:
`"data": {
      "unknown": false,
      "confidence": 100,
      "local_dt": {
        "minute": 0,
        "hour": 0,
        "second": 20,
        "day": 15,
        "weekday": 1,
        "month": 1,
        "timezone": "America/Los_Angeles",
        "year": 2019
      },
      "automotive": false,
      "cycling": false,
      "stationary": false,
      "running": false,
      "type": 9,
      "fmt_time": "2019-01-15T00:00:20.018576-08:00",
      "confidence_level": "high",
      "ts": 1547539220.0185764,
      "walking": false
    }`

**Comment by shankari** [comment]
By searching for "pipeline not working", you can find the prior such bug (#283)

**Comment by ecvick85** [comment]
Trevor sent me his logs from 1/31 - current date (all uncleaned trips). From
[https://github.com/e-mission/e-mission-docs/blob/a7d650349583f0dd35bfd7edf55bca784491f413/docs/e-mission-phone/troubleshooting_tips_faq.md#L134](url)
I was able to add date/time to his trips (but it didn't matter because none were cleaned).

Looking at the prior bugs (#283), I looked into webserver.log for Trevor's trips. Here's a section of the logs with Trevor's UID (I replaced his UID with fe1ca... and deleted the token list)

```
2019-02-28 03:24:45,934:DEBUG:140373085918976:START POST /timeline/getTrips/2019-02-27
2019-02-28 03:24:45,934:DEBUG:140373043955456:methodName = token_list, returning <class 'emission.net.auth.token_list.TokenListMethod'>
2019-02-28 03:24:45,934:DEBUG:140373085918976:Called timeline.getTrips/2019-02-27
2019-02-28 03:24:45,935:DEBUG:140373085918976:methodName = token_list, returning <class 'emission.net.auth.token_list.TokenListMethod'>
2019-02-28 03:24:45,935:DEBUG:140373043955456:Using the TokenListMethod to verify id token ... of length 8 against list [...]...
2019-02-28 03:24:45,935:DEBUG:140373043955456:Found match for token GjgJfa8p of length 8
2019-02-28 03:24:45,935:DEBUG:140373085918976:Using the TokenListMethod to verify id token ... of length 8 against list ['...']...
2019-02-28 03:24:45,936:DEBUG:140373085918976:Found match for token GjgJfa8p of length 8
2019-02-28 03:24:45,936:DEBUG:140373043955456:retUUID = fe1ca...
2019-02-28 03:24:45,937:DEBUG:140373085918976:retUUID = fe1ca...
2019-02-28 03:24:45,937:DEBUG:140373085918976:user_uuid fe1ca...
2019-02-28 03:24:45,937:DEBUG:140373085918976:Getting geojson for LocalDate({'year': 2019, 'month': 2, 'day': 27}) -> LocalDate({'year': 2019, 'month': 2, 'day': 27})
2019-02-28 03:24:45,937:INFO:140373085918976:About to query for date components LocalDate({'year': 2019, 'month': 2, 'day': 27}) -> LocalDate({'year': 2019, 'month': 2, 'day': 27})
2019-02-28 03:24:45,937:DEBUG:140373085918976:for user fe1ca..., returning timeseries <emission.storage.timeseries.builtin_timeseries.BuiltinTimeSeries object at 0x7fab0568aef0>
2019-02-28 03:24:45,938:DEBUG:140373085918976:In get_range_query, returning query {'data.enter_local_dt.year': {'$gte': 2019, '$lte': 2019}, 'data.enter_local_dt.month': {'$gte': 2, '$lte': 2}, 'data.enter_local_dt.day': {'$gte': 27, '$lte': 27}}
2019-02-28 03:24:45,938:DEBUG:140373085918976:curr_query = {'invalid': {'$exists': False}, 'user_id': UUID('fe1ca...'), '$or': [{'metadata.key': 'analysis/cleaned_place'}], 'data.enter_local_dt.year': {'$gte': 2019, '$lte': 2019}, 'data.enter_local_dt.month': {'$gte': 2, '$lte': 2}, 'data.enter_local_dt.day': {'$gte': 27, '$lte': 27}}, sort_key = data.enter_ts
2019-02-28 03:24:45,938:DEBUG:140373085918976:orig_ts_db_keys = [], analysis_ts_db_keys = ['analysis/cleaned_place']
2019-02-28 03:24:45,945:DEBUG:140373043955456:Returning complete_ts = None
2019-02-28 03:24:45,945:DEBUG:140373043955456:END POST /pipeline/get_complete_ts fe1ca... 0.011472463607788086
```
Comparing his trips to one from a user with a working pipeline, I noticed after his `START POST`
he doesn't have any calls to `/usercache/get` or '/usercache/put' like working pipeline users do (example of what working pipeline user's logs look like after `START POST` below). He only has calls to `/result/metrics/timestamp` and `/profile/update`. Could this be a problem?

```
2019-02-28 03:55:41,412:DEBUG:140373052348160:START POST /usercache/get
2019-02-28 03:55:41,413:DEBUG:140373052348160:Called userCache.get
...
2019-02-28 03:50:01,470:DEBUG:140373077526272:START POST /usercache/put
2019-02-28 03:50:01,470:DEBUG:140373077526272:Called userCache.put
```

Also, from what I found (pattern matching with the rest of the groups trips), Trevor doesn't have any lines associated with his uid like these (example from a user, replaced UID with ---, with working pipeline logs)

```
2019-02-28 03:50:01,708:DEBUG:140373077526272:Updated result for user = ---, key = background/motion_activity, write_ts = 1550688571.7149563 = {'n': 1, 'nModified': 0, 'ok': 1.0, 'updatedExisting': True}
...
2019-02-28 03:50:01,590:DEBUG:140373077526272:Updated result for user = ---, key = background/battery, write_ts = 1551325798.4757938 = {'n': 1, 'nModified': 0, 'upserted': ObjectId('5c775a69a73047157069c6ca'), 'ok': 1.0, 'updatedExisting': False}
```

I think that these are just background data of points on trips (please correct me if I am incorrect) - interesting that Trevor doesn't seem to have any in the logs.

Lastly, I looked at the query for his inferred trips (which don't exist) - and can be seen as nonexistent in the code too.

```
2019-02-28 03:22:59,217:DEBUG:140373052348160:curr_query = {'invalid': {'$exists': False}, 'user_id': UUID('fe1ca...'), '$or': [{'metadata.key': 'analysis/inferred_section'}], 'data.start_ts': {'$lte': 1551225600, '$gte': 1550102400}}, sort_key = data.start_ts
```

If you have any advice in log reading or anything stands out to you as being problematic that I missed - any help would be greatly appreciated.

**Comment by shankari** [comment]
> he doesn't have any calls to `/usercache/get` or `/usercache/put` like working pipeline users do (example of what working pipeline user's logs look like after `START POST` below). He only has calls to `/result/metrics/timestamp` and `/profile/update`. Could this be a problem?

Yes, this is the problem. As you can see from the troubleshooting steps
https://github.com/e-mission/e-mission-docs/blob/master/docs/e-mission-server/troubleshooting_tips_FAQ.md#is-the-data-getting-to-the-server-correctly
it looks like the data is not getting from his phone to the server.

So you now have the look at the phone logs to see what is going on.
The phone should be initiating `/usercache/get` and `/usercache/put` calls to the server periodically (~ 1 hour). What do you see around those calls? Anything similar to https://github.com/e-mission/e-mission-docs/issues/283#issuecomment-450407817?

**Comment by ecvick85** [comment]
Since the phone logs were only from his uncleaned trips, I don't see any `/usercache/get` or `/usercache/put` calls in them. There were no HTTP errors resulted from the calls (because they never happened) like in #283.

**Comment by shankari** [comment]
I am not sure why the phone logs would be different based on cleaned or uncleaned trips. 

1. Can you post the results of searching through the logs? Use the command line - e.g. `grep`
1. Can you post some privacy-preserving or redacted logs so that I can see that you have the right logs...

**Comment by ecvick85** [comment]
1. After searching through logs using grep for `/usercache/` (looking for either `/usercache/put` or `/usercache/get`, nothing was outputted --> there is no sign of `/usercache/` in Trevor's logs:
```
Emmas-MacBook-Pro-3:e-mission-phone emmavickery$ grep "/usercache/" TrevorLogs.withdate.log
Emmas-MacBook-Pro-3:e-mission-phone emmavickery$ 
```
2. Here's a snippet of the logs I was sent:
```
0,1548834584.734,2019-01-29 23:49:44.734000-07:53,js : Finished clearing entries from unified log
1,1548834586.238,2019-01-29 23:49:46.238000-07:53,BuiltinUserCache : Added value for key stats/client_time at time 1.548834586238E9
2,1548834586.244,2019-01-29 23:49:46.244000-07:53,"js : uc_stored_val = {""intro_done"":""2019-01-20T17:19:59-08:00""} ls_stored_val = {""intro_done"":""2019-01-20T17:19:59-08:00""}"
3,1548834586.245,2019-01-29 23:49:46.245000-07:53,"js : local and native values match, already synced"
4,1548834586.245,2019-01-29 23:49:46.245000-07:53,"js : in readIntroDone, read_val = ""2019-01-20T17:19:59-08:00"""
5,1548834586.245,2019-01-29 23:49:46.245000-07:53,"js : in isIntroDone, returning true"
6,1548834586.26,2019-01-29 23:49:46.260000-07:53,"js : required consent version = {""protocol_id"":""2014-04-6267"",""approval_date"":""2016-07-14""}"
7,1548834586.2779999,2019-01-29 23:49:46.278000-07:53,"js : uc_stored_val = {""protocol_id"":""2014-04-6267"",""approval_date"":""2016-07-14""} ls_stored_val = {""protocol_id"":""2014-04-6267"",""approval_date"":""2016-07-14""}"
8,1548834586.2779999,2019-01-29 23:49:46.278000-07:53,"js : local and native values match, already synced"
9,1548834586.553,2019-01-29 23:49:46.553000-07:53,"ConnectionSettings : in getSettings, connectionSettings = {""connectUrl"":""http:\/\/52.91.65.149:8080"",""android"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}},""ios"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}}}"
10,1548834586.658,2019-01-29 23:49:46.658000-07:53,"js : uc_stored_val = {""deploy_channel"":""greentrip""} ls_stored_val = {""deploy_channel"":""greentrip""}"
11,1548834586.66,2019-01-29 23:49:46.660000-07:53,"js : local and native values match, already synced"
12,1548834586.664,2019-01-29 23:49:46.664000-07:53,js : uc_stored_val = null ls_stored_val = null
13,1548834586.665,2019-01-29 23:49:46.665000-07:53,"js : local and native values match, already synced"
14,1548834587.085,2019-01-29 23:49:47.085000-07:53,BuiltinUserCache : Added value for key stats/client_time at time 1.548834587085E9
15,1548834587.0970001,2019-01-29 23:49:47.097000-07:53,"ConnectionSettings : in getConnectURL, connectionSettings = {""connectUrl"":""http:\/\/52.91.65.149:8080"",""android"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}},""ios"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}}}"
16,1548834587.0970001,2019-01-29 23:49:47.097000-07:53,"ConnectionSettings : in getConnectURL, returning http://52.91.65.149:8080"
17,1548834587.0970001,2019-01-29 23:49:47.097000-07:53,"ConnectionSettings : in getConnectURL, connectionSettings = {""connectUrl"":""http:\/\/52.91.65.149:8080"",""android"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}},""ios"":{""auth"":{""method"":""prompted-auth"",""prompt"":""Please enter valid token:""}}}"
18,1548834587.0970001,2019-01-29 23:49:47.097000-07:53,"ConnectionSettings : in getConnectURL, returning http://52.91.65.149:8080"
19,1548834587.1279998,2019-01-29 23:49:47.128000-07:53,"js : uc_stored_val = {""intro_done"":""2019-01-20T17:19:59-08:00""} ls_stored_val = {""intro_done"":""2019-01-20T17:19:59-08:00""}"
20,1548834587.1279998,2019-01-29 23:49:47.128000-07:53,"js : local and native values match, already synced"
```

**Comment by shankari** [comment]
@ecvick85 I think you have done enough initial debugging :)
This doesn't appear to be a known problem. Please send me the log over email and I'll take over from here.

Good job!

**Comment by shankari** [comment]
I believe this was resolved after the latest update that includes #325
